NebulaWorld/MonoBehaviours/Remote/RemotePlayerMovement.cs

```diff
@@ -112,6 +120,31 @@ private void MoveInterpolated(Snapshot previous, Snapshot current, float ratio)
         {
             Vector3 previousPosition = GetRelativePosition(previous);
             Vector3 currentPosition = GetRelativePosition(current);
+            float deltaPosition = Vector3.Distance(previousPosition, currentPosition);
+            Vector3 velocity = (previousPosition - currentPosition) / (previous.Timestamp - current.Timestamp);
+
+            // needed as its null for some reason when we arrive here for the first time
+            if(rootWarp == null)
+            {
+                rootWarp = rootTransform.GetComponent<RemoteWarpEffect>();
+            }
```

Comment:
Try renaming the `Awake()` method to `Start()` method instead you should not have initialization race condition anymore.

`Awake()` in your case is called as soon as you add the component to the gameObject.
`Start()` is called on the frame when a script is enabled just before any of the Update methods are called the first time.